### ISSUE TITLE
Providing default values to flags, e.g. `--packages=bundle`

### ISSUE THREAD

**Issue by mrgrain** [evanw/esbuild#3824]
❗ **This issue deliberately does not have an immediate action. It's more of a discussion or question.**

During the  `v0.21.0` /  `v0.22.0` / `v0.23.0` updates cycle, I noticed that that `--packages=bundle` does not work on versions `<= 0.21.0` despite the flag being introduced in `v.0.16.0`. Passing `bundle` started working in. `v0.22.0` because the default changed to `external`. And the revert of the default in `v0.23.0` kept the ability to provide either value. 

Now this might have just been an outlier and in general passing default values to options just works. Really what I want to highlight here is that I think it's quite important to be able to pass in a default.

Some background here: I maintain an integration package that is effectively a thin wrapper around `esbuild`. I use similar disclaimers regarding stability and breaking changes. I also encourage users to pin their versions. However as a wrapper library, I will make _different choices for defaults_ because they just make more sense in the context of the ecosystem my library is integrating with. I'm happy to do this, and I'm happy to maintain these extra defaults and test them carefully with each new version. But it does make the life of my users and my harder, if for example I cannot provide the default value of a flag to safeguard against future default changes in esbuild (or - admittedly - to react to them). 

There's obviously a balance to be struck here. For example I wouldn't really expect the `--packages` flag to be "usable" (but without effect) before it existed in `v.0.16.0`. 

### What's the call to action?

There is none. The only situation that I am aware of where this was not allowed previously has now been fixed. I also checked the TS types as a proxy and couldn't find any other case.

So this issue really just serves as a request, and FYI that this is something that's important to a (small) subset of your community.

### Reproduction

```console
$ touch index.js

$ npx esbuild@0.21.x index.js --packages=bundle
✘ [ERROR] Invalid value "bundle" in "--packages=bundle"

  The only valid value is "external".

1 error

$ npx esbuild@0.22.x index.js --packages=bundle

$ npx esbuild@0.23.x index.js --packages=bundle
```

**Comment by evanw** [comment]
I agree. I think this was an outlier, and I'll try to keep this in mind when adding flags in the future. I'm not aware of any other cases of this.

FWIW if you're using the esbuild library API, there is a [`version` field](https://github.com/evanw/esbuild/blob/9d506806bdd963b02b3d6edf45e717e03dcba785/lib/shared/types.ts#L665) that you can use to easily tell what esbuild version you're dealing with.

**Comment by mrgrain** [comment]
Awesome. I'm happy we are on the same page regarding this. 

I had noticed the `versions` flag. Unfortunately it wasn't as simple as adding a `condition` due to the architecture on my end (tl;dr loading esbuild is delayed to the last possible moment), and then you had already released the revert. But that's definitely something I'll be looking into as it will make my design much more resilient.
